### PR TITLE
Update the service to take transactions from the blockchain [ECR-2595]

### PR DESCRIPTION
**Pull request by dmitry-timofeev**
## Overview

Update the service to use transaction messages preserved
in the blockchain to collect a history of operations
of a particular wallet instead of storing duplicate
history entries in the service storage.

---
See: https://jira.bf.local/browse/ECR-2595

### Definition of Done

- [x] There are no TODOs left in the code
- [x] Change is covered by automated [tests](https://github.com/exonum/exonum-java-binding/blob/master/CONTRIBUTING.md#tests)
- [x] The [coding guidelines](https://github.com/exonum/exonum-java-binding/blob/master/CONTRIBUTING.md#the-code-style) are followed
- [x] Public API has Javadoc
- [x] Method preconditions are checked and documented in the Javadoc of the method
- [x] Changelog is updated if needed (in case of notable or breaking changes)
- [ ] The continuous integration build passes

### CODE REVIEW

File: exonum-java-binding-cryptocurrency-demo/src/main/java/com/exonum/binding/cryptocurrency/CryptocurrencyServiceImpl.java [hunk 1]
```diff
@@ -82,17 +88,36 @@ public void createPublicApiHandlers(Node node, Router router) {
...
-
-      List<HistoryEntity> result = new ArrayList<>();
-      for (HistoryEntity tx : history) {
-        result.add(tx);
-      }
-
-      return result;
+      ListIndex<HashCode> walletHistory = schema.transactionsHistory(ownerKey);
+      Blockchain blockchain = Blockchain.newInstance(view);
+      MapIndex<HashCode, TransactionMessage> txMessages = blockchain.getTxMessages();
```
Comment (dmitry-timofeev) [inline exonum-java-binding-cryptocurrency-demo/src/main/java/com/exonum/binding/cryptocurrency/CryptocurrencyServiceImpl.java]: With no testkit it isn't possible to test these operations easily. I _could_ have put some data into `txMessages` in test setup code, but … we had decided on a readonly `Blockchain`, hadn't we?

File: exonum-java-binding-cryptocurrency-demo/src/main/java/com/exonum/binding/cryptocurrency/CryptocurrencyServiceImpl.java [hunk 2]
```diff
@@ -82,15 +89,39 @@ public void createPublicApiHandlers(Node node, Router router) {
...
+    return StreamSupport.stream(list.spliterator(), false);
+  }
 
-      return result;
-    });
+  private HistoryEntity createTransferHistoryEntry(TransactionMessage txMessage) {
+    checkState(txMessage.getServiceId() == getId(),
+        "service id mismatch: message contains %s, expected %s", txMessage.getServiceId(), getId());
+    try {
+      TxMessageProtos.TransferTx txBody = TxMessageProtos.TransferTx
```
Comment (bullet-tooth) [inline exonum-java-binding-cryptocurrency-demo/src/main/java/com/exonum/binding/cryptocurrency/CryptocurrencyServiceImpl.java]: `StandardSerializers#protobuf`?

File: exonum-java-binding-cryptocurrency-demo/src/main/java/com/exonum/binding/cryptocurrency/CryptocurrencyServiceImpl.java [hunk 3]
```diff
@@ -82,15 +89,39 @@ public void createPublicApiHandlers(Node node, Router router) {
...
+  private static Stream<HashCode> stream(ListIndex<HashCode> list) {
+    // TODO: Replace with a proper thing in ECR-597!
+    return StreamSupport.stream(list.spliterator(), false);
+  }
 
-      return result;
-    });
+  private HistoryEntity createTransferHistoryEntry(TransactionMessage txMessage) {
+    checkState(txMessage.getServiceId() == getId(),
+        "service id mismatch: message contains %s, expected %s", txMessage.getServiceId(), getId());
```
Comment (MakarovS) [inline exonum-java-binding-cryptocurrency-demo/src/main/java/com/exonum/binding/cryptocurrency/CryptocurrencyServiceImpl.java]: `Service ID mismatch: ...`

File: exonum-java-binding-cryptocurrency-demo/src/main/java/com/exonum/binding/cryptocurrency/CryptocurrencyServiceImpl.java (hunk 2 above)
Comment (dmitry-timofeev) [reply exonum-java-binding-cryptocurrency-demo/src/main/java/com/exonum/binding/cryptocurrency/CryptocurrencyServiceImpl.java]: I'd rather not — it requires a separate constant static field :-)

**Pull request merged**